bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/contentassist/IContentAssistantExtension5.java

```diff
@@ -0,0 +1,29 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Dawid Pakuła and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Dawid Pakuła - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jface.text.contentassist;
+
+/**
+ * Extends {@link org.eclipse.jface.text.contentassist.IContentAssistant} with the following
+ * function:
+ * <ul>
+ * <li>allows to build IContentAssitRequest</li>
+ * </ul>
+ *
+ * @since 3.24
+ */
+public interface IContentAssistantExtension5 {
+
+	IContentAssistRequest buildAssistantRequest(int offset, boolean isAutoAssit, boolean isIncremental);
```

Comment:
Can't this be internal to ContentAssistant?

Reply:
According to defaults, without them if want switch to IContentAssistRequest methods only, you still have to build IContentAssistProcessor with old one